assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js

```diff
@@ -21,31 +21,101 @@
  */
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
 import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import {
+	MetricTileTable,
+	whenKeyMetricsWidgetVisible,
+} from '../../../../components/KeyMetrics';
+import Link from '../../../../components/Link';
+import { ZeroDataMessage } from '../../../analytics/components/common';
+import { numFmt } from '../../../../util';
+const { useSelect, useInViewSelect } = Data;
+
+function PopularContentWidget( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
 
-const { useSelect } = Data;
+	const reportOptions = {
+		...dates,
+		dimensions: [ 'pageTitle', 'pagePath' ],
+		metrics: [ { name: 'screenPageViews' } ],
+		orderby: [
+			{
+				metric: { metricName: 'screenPageViews' },
+				desc: true,
+			},
+		],
+		limit: 3,
+	};
 
-export default function PopularContentWidget( { Widget, WidgetNull } ) {
-	const keyMetricsWidgetHidden = useSelect( ( select ) =>
-		select( CORE_USER ).isKeyMetricsWidgetHidden()
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( reportOptions )
 	);
 
-	if ( keyMetricsWidgetHidden !== false ) {
-		return <WidgetNull />;
```

Comment:
Hmm, I notice this widget now doesn't return null, but the IB does specify that it _should_ return null under a couple of circumstances:

  * Return `null` if the `GA4` module is not connected.
  * Using the `getKeyMetrics` selector, check if any of the selected Metric is from GA4. If not, return `null`. Otherwise, continue.

Now, I have also spotted that the QAB says "For now, all widgets are displayed on the dashboard (which will change with future commits)", however the only relevant issue I can see is https://github.com/google/site-kit-wp/issues/7061, which should take care of the second point relating to `getKeyMetrics`; the point about returning `null` (actually, `WidgetNull` would be more appropriate) if GA4 is not connected still seems like it's probably relevant. We are already doing something similar in at least one of the other Key Metrics widgets, the `ConnectGA4CTATileWidget`:

https://github.com/google/site-kit-wp/blob/d7bfa9887ff7e6f908c1d6c3f173dc68b22bd975/assets/js/modules/analytics-4/components/widgets/ConnectGA4CTATileWidget.js#L62-L64

So, to summarise, it looks like returning `WidgetNull` when GA4 is not connected is probably still the right thing to do here, whereas the point about checking the `getKeyMetrics` selector will effectively be taken care of later via #7061. What do you think?